python/ray/_raylet.pyx

```diff
@@ -903,17 +906,52 @@ cdef class CoreWorker:
             extra_data)
 
     def deserialize_and_register_actor_handle(self, const c_string &bytes):
+        from ray.actor import ActorHandle, ActorClassMethodMetadata
+        cdef CActorHandle* c_actor_handle
+        worker = ray.worker.get_global_worker()
+        worker.check_connected()
+        manager = worker.function_actor_manager
         c_actor_id = self.core_worker.get().DeserializeAndRegisterActorHandle(
             bytes)
+        check_status(self.core_worker.get().GetActorHandle(
+            c_actor_id, &c_actor_handle))
         actor_id = ActorID(c_actor_id.Binary())
-        return actor_id
-
-    def serialize_actor_handle(self, ActorID actor_id):
+        job_id = JobID(c_actor_handle.CreationJobID().Binary())
+        language = Language.from_native(c_actor_handle.ActorLanguage())
+        actor_creation_function_descriptor = \
+            CFunctionDescriptorToPython(
+                c_actor_handle.ActorCreationTaskFunctionDescriptor())
+        if language == Language.PYTHON:
+            assert isinstance(actor_creation_function_descriptor,
+                              PythonFunctionDescriptor)
+            actor_method_cpu = int(c_actor_handle.ExtensionData())
+            actor_class = manager.load_actor_class(
+                job_id, actor_creation_function_descriptor)
+            method_meta = ActorClassMethodMetadata(actor_class)
+            return ActorHandle(language, actor_id,
+                               method_meta.decorators,
+                               method_meta.signatures,
+                               method_meta.num_return_vals,
+                               actor_method_cpu,
+                               actor_creation_function_descriptor,
+                               worker.current_session_and_job)
+        else:
+            return ActorHandle(language, actor_id,
+                               {},  # method decorators
+                               {},  # method signatures
+                               {},  # method num_return_vals
+                               0,  # actor method cpu
+                               actor_creation_function_descriptor,
+                               worker.current_session_and_job)
+
+    def serialize_actor_handle(self, actor_handle):
+        from ray.actor import ActorHandle
```

Comment:
I don't know if it's a good idea to import here.

Reply:
Import inside function to avoid cyclic imports.